### PR TITLE
Add an e2e test for third party resources (disabled by default)

### PR DESCRIPTION
**Pull request by brendandburns**
@ixdy @spxtr @krousey

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -285,6 +285,7 @@ GKE_REQUIRED_SKIP_TESTS=(
     # Alpha features, remove from skip when these move to beta
     "Daemon\sset"
     "Deployment"
+    "ThirdParty"
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: @ihmccreery is trying to get rid of this list

Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Yes, please change this instead to use `SkipIfProviderIs` or `SkipUnlessProviderIs`.

File: test/e2e/third-party.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	unversioned.ListMeta `json:"metadata,omitempty" description:"standard list metadata; see http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata"`
+
+	Items []Foo `json:"items"`
+}
+
+var _ = Describe("ThirdParty resources", func() {
+
+	f := NewFramework("thirdparty")
+
+	rsrc := &extensions.ThirdPartyResource{
```
Comment (lavalamp) [inline test/e2e/third-party.go]: did you mean to unmarshal the `data` blob above?

File: test/e2e/third-party.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	"encoding/json"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var data = `{
```
Comment (lavalamp) [inline test/e2e/third-party.go]: recommend a more specific name to avoid polluting the shared namespace.

File: test/e2e/third-party.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+			defer func() {
+				if err := f.Client.ThirdPartyResources("").Delete(rsrc.Name); err != nil {
+					Failf("failed to delete third party resource: %v", err)
+				}
+			}()
+			if _, err := f.Client.ThirdPartyResources("").Create(rsrc); err != nil {
+				Failf("failed to create third party resource: %v", err)
+			}
+
+			data, err := f.Client.RESTClient.Get().AbsPath("/apis/company.com/v1/foos").DoRaw()
```
Comment (lavalamp) [inline test/e2e/third-party.go]: I don't think `company.com` is in `rsrc`, so I don't see how this can work?

File: test/e2e/third-party.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+			Name: "test",
+		},
+		Versions: []extensions.APIVersion{
+			{Name: "v1"},
+		},
+	}
+
+	Context("Simple Third Party", func() {
+		It("creating/deleting thirdparty objects works [Conformance]", func() {
+			defer func() {
```
Comment (lavalamp) [inline test/e2e/third-party.go]: Maybe move this defer below where you actually create it?

**Comment by lavalamp** [comment]
looks good but a few comments

**Comment by lavalamp** [comment]
@brendandburns Are you going to pick this up again?

**Comment by lavalamp** [comment]
@brendandburns Please reopen when you have time.

**Comment by brendandburns** [comment]
@lavalamp this is ready for (re) review.

Depends on https://github.com/kubernetes/kubernetes/pull/25365

**Comment by brendandburns** [comment]
There's something weird about this PR, closing it in favor of:

https://github.com/kubernetes/kubernetes/pull/25542

**Pull request closed without merging**